features/include/pcl/features/impl/integral_image_normal.hpp

```diff
@@ -107,12 +107,12 @@ pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::initSimple3DGradientMet
   // number of DataType entries per element (equal or bigger than dimensions)
   int element_stride = sizeof (PointInT) / sizeof (float);
   // number of DataType entries per row (equal or bigger than element_stride number of elements per row)
-  int row_stride     = element_stride * input_->width;
+  int row_stride     = element_stride * static_cast<uindex_t>(input_->width);
```

Comment:
Is defining `element_stride` and `row_stride` as `index_t` an option?

Reply:
These strides are passed to `integral_image_XYZ_.setInput (const DataType * data, unsigned width,unsigned height, unsigned element_stride, unsigned row_stride)` which takes in unsigned values for strides. Would it be a good idea to deprecate this function with the SFINAE approach? This function is called multiple times, and requires a lot of casts.

Reply:
Yes, definitely. Those are the best cases to deprecate.